src/problem/problem.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { ProblemRepository } from '../database/repositories/problem.repository';
+import { Problem } from '../database/schemas/problem.schema';
+import { Comment } from '../database/schemas/comment.schema';
+
+@Injectable()
+export class ProblemService {
+    constructor(private readonly problemRepository: ProblemRepository) {}
+
+    public async createProblem(problem: Problem): Promise<Problem> {
+        const newProblem = new this.problemRepository.GetProblemModel(problem);
+
+        return await newProblem.save();
+    };
+
+    public async getAllProblems(): Promise<Problem[]> {
+        const problems = await this.problemRepository.GetProblemModel.find();
+
+        if (!problems || problems.length === 0) {
+            Logger.warn('Nenhum problema cadastrado', 'ProblemService');
+
+            throw new NotFoundException('Nenhum problema cadastrado');
```

Comment:
Não precisa lançar exceção, só retorna o array vazio mesmo.
```suggestion
                  return problems;
```